s3torchconnectorclient/rust/src/logger_setup.rs

```
@@ -0,0 +1,117 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * // SPDX-License-Identifier: BSD
+ */
+use std::{env};
+use log::{LevelFilter};
+use mountpoint_s3_crt::common::rust_log_adapter::RustLogAdapter;
+use pyo3::{PyResult};
+use pyo3_log::Logger;
+use tracing_subscriber::{filter::EnvFilter};
+use tracing_subscriber::util::{SubscriberInitExt};
+use crate::exception::python_exception;
+
+pub const ENABLE_CRT_LOGS_ENV_VAR: &str = "ENABLE_CRT_LOGS";
+pub const CRT_LOGS_DIR_PATH_ENV_VAR: &str = "CRT_LOGS_DIR_PATH";
+
+pub fn setup_logging() -> PyResult<()> {
+    let enable_crt_logs = env::var(ENABLE_CRT_LOGS_ENV_VAR).ok();
+
+    match enable_crt_logs {
+        None => enable_default_logging(),
+        Some(..) => enable_crt_logging()
+    }
+}
+
+fn enable_crt_logging() -> PyResult<()> {
+    RustLogAdapter::try_init().map_err(python_exception)?;
+
+    let filter = EnvFilter::try_from_env(ENABLE_CRT_LOGS_ENV_VAR).map_err(python_exception);
```

Similarly to James' comments in the last PR, this is swallowing the error